src/EditorFeatures/Core/Implementation/MissingImportTracking/MissingImportTrackingTaggerProvider.MissingImportTrackingCommitter.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddImport;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.MissingImportTracking
+{
+    internal partial class MissingImportTrackingTaggerProvider
+    {
+        private class MissingImportTrackingCommitter
+        {
+            private StateMachine StateMachine { get; }
+            private Document Document { get; }
+
+            public MissingImportTrackingCommitter(StateMachine stateMachine, Document document, string title)
+            {
+                StateMachine = stateMachine;
+                Document = document;
+            }
+
+            public async Task Commit(CancellationToken cancellationToken)
+            {
+                var updatedProject = await AddMissingImportsAsync(cancellationToken).ConfigureAwait(false);
+                var workspace = Document.Project.Solution.Workspace;
+
+                if (workspace.TryApplyChanges(updatedProject.Solution))
+                {
+                    StateMachine.ClearTrackingSpans();
```

Comment:
is this mean this and rename can't be used at the same time? probably okay since if one is invoked, it changes document and tracking is dismissed. just want to make sure that this state tracking only support one session at a time for all features that uses the state tracking.

Reply:
You can use both. The StateMachines are separate implementations. I can update class names to differentiate the two better.